### ISSUE TITLE
IDE0040 warning 

### ISSUE THREAD

**Issue by CoolDadTx** [dotnet/roslyn#36544]
Visual Studio 2019 16.2.0 Preview 2.0

IDE0040 warning generated on partial classes even if one of the partial class declarations have one.

Steps:
1. Create a partial class definition with an access modifier for the type.
2. Create a partial class definition for the same type without the modifier.

```csharp
public static partial class ContainerConfig
{}

static partial class ContainerConfig   //IDE0040
{ }
```

Expected Behavior:
This is debatable but the general rule of thumb (from my observation) for partials is to only specify the type (class/struct) and name of the partial class in all but one declaration. This reduces the likelihood of making a mistake that conflicts with one of the other declarations. The language will use whichever one has the access modifier and therefore, while it is not explicitly set on all the partial declarations, it is inferred from the partial declaration that has it.

Actual Behavior:
The partial declarations that don't have the modifier are flagged with a warning.

**Comment by sharwell** [comment]
Design review conclusion: We believe the current behavior best reflects the intent of maintainers enabling IDE0040. To address the specific usability constraint relating to changing the access modifier for a type with multiple parts, we are filing a follow-up issue to create a code fix to consolidate the access modifiers for a type with inconsistent parts.

**Comment by ericwj** [comment]
I argue that specifying basically anything twice is contra to the design of C#.

Also the situation with partial declarations is so different from the normal occurrence of `IDE0040` that if you care about having access modifiers you cannot disable this warning and you are left with getting ambiguous fix suggestions (which access modifier was correct?) or the mandatory bloat like this:

```CS
#pragma warning disable IDE0040 // Add accessibility modifiers
    partial class Class
#pragma warning restore IDE0040 // Add accessibility modifiers
    {
        
    }
```

Can we have another IDE code for this situation that can be suppressed with `<NoWarn>` or a setting in `.editorconfig` please?

**Comment by sharwell** [comment]
@ericwj IDE0040 implements one approach to this code style rule, but different teams may want other approaches. Since Roslyn has an extensibility API, there are several options available to help with this:

1. You could create a new analyzer which is referenced by your projects, which implements the desired code style as a self contained rule. IDE0040 would then be disabled to avoid duplicate diagnostics.
2. You could use the new analyzer suppression API (a "diagnostic suppressor") to identify locations of IDE0040 which do not need to be fixed, and programmatically suppress the diagnostics at those locations.

**Comment by ericwj** [comment]
I'd rather have this inbox and enabled depending on the `.editorconfig` that I keep around but thanks for the suggestions.

**Comment by jnm2** [comment]
> We believe the current behavior best reflects the intent of maintainers enabling IDE0040

- I didn't run into IDE0040 as a result of maintainers enabling it. I ran into it as a result of .NET 5 applying it to everybody by default.
- I'm a maintainer that would like to use IDE0040 and I never want it to apply in this situation.
